commit 7409cd1cd554a0bdedfecf2afde58ff468b7045e
Author: Sascha Hauer <s.hauer@pengutronix.de>
Date:   Wed Aug 24 08:54:04 2011 +0200

    ARM i.MX: allow to compile together all i.MX5 based SoCs
    
    For this we need CONFIG_AUTO_ZRELADDR and CONFIG_ARM_PATCH_PHYS_VIRT.
    
    Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>

Index: linux-3.1/arch/arm/mach-mx5/Kconfig
===================================================================
--- linux-3.1.orig/arch/arm/mach-mx5/Kconfig	2011-11-13 14:24:52.090827440 +0100
+++ linux-3.1/arch/arm/mach-mx5/Kconfig	2011-11-13 14:24:52.158827438 +0100
@@ -1,8 +1,9 @@
-if ARCH_MX503 || ARCH_MX51
+if ARCH_MX5
+
 # ARCH_MX5/50/53 are left to mark places where prevent multi-soc in single
 # image. So for most time, SOC_IMX50/51/53 should be used.
 
-config ARCH_MX5
+config ARCH_MX51
 	bool
 
 config ARCH_MX50
@@ -19,7 +20,6 @@ config SOC_IMX50
 	select ARCH_MXC_IOMUX_V3
 	select ARCH_MXC_AUDMUX_V2
 	select ARCH_HAS_CPUFREQ
-	select ARCH_MX5
 	select ARCH_MX50
 
 config	SOC_IMX51
@@ -30,7 +30,7 @@ config	SOC_IMX51
 	select ARCH_MXC_IOMUX_V3
 	select ARCH_MXC_AUDMUX_V2
 	select ARCH_HAS_CPUFREQ
-	select ARCH_MX5
+	select ARCH_MX51
 
 config	SOC_IMX53
 	bool
@@ -38,10 +38,8 @@ config	SOC_IMX53
 	select ARM_L1_CACHE_SHIFT_6
 	select MXC_TZIC
 	select ARCH_MXC_IOMUX_V3
-	select ARCH_MX5
 	select ARCH_MX53
 
-if ARCH_MX50_SUPPORTED
 #comment "i.MX50 machines:"
 
 config MACH_MX50_RDP
@@ -57,9 +55,6 @@ config MACH_MX50_RDP
 	  Include support for MX50 reference design platform (RDP) board. This
 	  includes specific configurations for the board and its peripherals.
 
-endif # ARCH_MX50_SUPPORTED
-
-if ARCH_MX51
 comment "i.MX51 machines:"
 
 config MACH_MX51_BABBAGE
@@ -168,9 +163,6 @@ config MACH_MX51_EFIKASB
 	  Include support for Genesi Efika Smartbook. This includes specific
 	  configurations for the board and its peripherals.
 
-endif # ARCH_MX51
-
-if ARCH_MX53_SUPPORTED
 comment "i.MX53 machines:"
 
 config MACH_MX53_EVK
@@ -222,6 +214,4 @@ config MACH_MX53_ARD
 	  Include support for MX53 ARD platform. This includes specific
 	  configurations for the board and its peripherals.
 
-endif # ARCH_MX53_SUPPORTED
-
 endif
Index: linux-3.1/arch/arm/plat-mxc/Kconfig
===================================================================
--- linux-3.1.orig/arch/arm/plat-mxc/Kconfig	2011-11-13 14:20:20.026839411 +0100
+++ linux-3.1/arch/arm/plat-mxc/Kconfig	2011-11-13 14:24:52.158827438 +0100
@@ -4,12 +4,6 @@ source "arch/arm/plat-mxc/devices/Kconfi
 
 menu "Freescale MXC Implementations"
 
-config ARCH_MX50_SUPPORTED
-	bool
-
-config ARCH_MX53_SUPPORTED
-	bool
-
 choice
 	prompt "Freescale CPU family:"
 	default ARCH_MX3
@@ -34,20 +28,14 @@ config ARCH_MX3
 	help
 	  This enables support for systems based on the Freescale i.MX3 family
 
-config ARCH_MX503
-	bool "i.MX50 + i.MX53"
-	select ARCH_MX50_SUPPORTED
-	select ARCH_MX53_SUPPORTED
+config ARCH_MX5
+	bool "i.MX50, i.MX51, i.MX53"
+	select AUTO_ZRELADDR
+	select ARM_PATCH_PHYS_VIRT
 	help
 	  This enables support for machines using Freescale's i.MX50 and i.MX51
 	  processors.
 
-config ARCH_MX51
-	bool "i.MX51"
-	select ARCH_MX51_SUPPORTED
-	help
-	  This enables support for systems based on the Freescale i.MX51 family
-
 endchoice
 
 source "arch/arm/mach-imx/Kconfig"
Index: linux-3.1/arch/arm/plat-mxc/include/mach/memory.h
===================================================================
--- linux-3.1.orig/arch/arm/plat-mxc/include/mach/memory.h	2011-11-13 14:20:20.014839413 +0100
+++ linux-3.1/arch/arm/plat-mxc/include/mach/memory.h	2011-11-13 14:24:52.158827438 +0100
@@ -31,12 +31,6 @@
 #  define PLAT_PHYS_OFFSET		MX27_PHYS_OFFSET
 # elif defined CONFIG_ARCH_MX3
 #  define PLAT_PHYS_OFFSET		MX3x_PHYS_OFFSET
-# elif defined CONFIG_ARCH_MX50
-#  define PLAT_PHYS_OFFSET		MX50_PHYS_OFFSET
-# elif defined CONFIG_ARCH_MX51
-#  define PLAT_PHYS_OFFSET		MX51_PHYS_OFFSET
-# elif defined CONFIG_ARCH_MX53
-#  define PLAT_PHYS_OFFSET		MX53_PHYS_OFFSET
 # endif
 #endif
 
